src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -43,16 +38,17 @@ class RsPathReferenceImpl(
     override fun advancedResolve(): BoundElement<RsElement>? =
         advancedMultiResolve().firstOrNull()
 
-    override fun multiResolve(incompleteCode: Boolean): Array<out ResolveResult> =
-        advancedMultiResolve().toTypedArray()
-
-    override fun multiResolve(): List<RsNamedElement> =
-        advancedMultiResolve().mapNotNull { it.element as? RsNamedElement }
-
     override fun advancedMultiResolve(): List<BoundElement<RsElement>> =
         (element.parent as? RsPathExpr)?.let { it.inference?.getResolvedPath(it)?.map { BoundElement(it) } }
             ?: advancedCachedMultiResolve()
 
+    override fun multiResolve(incompleteCode: Boolean): Array<out ResolveResult> {
+        return (if (element.parentOfType<RsPatConst>() != null) resolvePath(element) else advancedMultiResolve()).toTypedArray()
+    }
+
+    override fun multiResolve(): List<RsNamedElement> =
+        multiResolve(false).mapNotNull { it.element as? RsNamedElement }
```

Comment:
`RsPathReferenceImpl` is way to complicated :( It would be great to try to simplify it a bit, together!